### PR TITLE
:sparkles:  :left_speech_bubble: CLM-48: Abiltiy to create business initiated messages

### PR DESCRIPTION
**Pull request by paulkahura**
# Description

Message templates frontend.

CLM-48: Ability to create and register busines initiated messages

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: libs/private/features/convs-mgr/message-templates/src/lib/components/message-template-list/message-template-list.component.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    });
+    
+  }
+  applyFilter(filterValue: string) {
+    filterValue = filterValue.trim().toLowerCase();
+    this.dataSource.filter = filterValue;
+  }
+
+  onCreate() {
+    throw new Error('Method not implemented.');
```
Comment (ReaganCn) [inline libs/private/features/convs-mgr/message-templates/src/lib/components/message-template-list/message-template-list.component.ts:71]: I thought this is already implemented

File: libs/private/features/convs-mgr/message-templates/src/lib/components/message-template-single-settings/message-template-single-settings.component.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+export class MessageTemplateSingleSettingsComponent implements OnInit{
+  selectedOption: string;
+  selectedTime: Date;
+  messageTemplateFrequency = [
+    { value: 'milestone', viewValue: 'Milestone reached' },
+    { value: 'specific-time', viewValue: 'Send message at specific time' },
+    { value: 'inactivity', viewValue: 'After inactivity' },
+  ];
+  action: string;
+  canBeScheduled = false;
```
Comment (ReaganCn) [inline libs/private/features/convs-mgr/message-templates/src/lib/components/message-template-single-settings/message-template-single-settings.component.ts:26]: Also no need to set it to false

File: libs/private/features/convs-mgr/message-templates/src/lib/components/message-template-single-settings/message-template-single-settings.component.ts [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+        inactivityDialogRef.componentInstance?.timeInHoursSelected.subscribe(
+          (selectedTime: number) => {
+            // Process the selected time in hours as needed
+            const specificTimeOption = this.messageTemplateFrequency.find(option => option.value === 'inactivity');
+            if (specificTimeOption) {
+              specificTimeOption.viewValue = `Send message after ${selectedTime} of inactivity.`;
+            }
+          }
+        );
```
Comment (ReaganCn) [inline libs/private/features/convs-mgr/message-templates/src/lib/components/message-template-single-settings/message-template-single-settings.component.ts:77]: You can declutter the switch statement by moving the case functionalities to different methods

File: libs/private/state/message-templates/src/lib/service/message-template.service.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
+import { Injectable } from '@angular/core';
+import { AngularFireFunctions } from '@angular/fire/compat/functions';
+
+import { Observable } from 'rxjs';
+import { MessageTemplateStore } from '../store/message-template.store';
```
Comment (ReaganCn) [inline libs/private/state/message-templates/src/lib/service/message-template.service.ts:5]: Order of imports

File: libs/private/state/message-templates/src/lib/store/message-template.store.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
+import { AngularFireFunctions } from '@angular/fire/compat/functions';
+import { Injectable } from '@angular/core';
+
+import { Repository, DataService } from '@ngfi/angular';
+import { DataStore } from '@ngfi/state';
+
+
+import { Logger } from '@iote/bricks-angular';
+import { tap, throttleTime, switchMap, map } from 'rxjs/operators';
+import { of } from 'rxjs';
```
Comment (ReaganCn) [inline libs/private/state/message-templates/src/lib/store/message-template.store.ts:10]: Order of imports

**Review by ReaganCn — CHANGES_REQUESTED** [review]

File: libs/model/convs-mgr/functions/src/lib/base/message-template.interface.ts [hunk 6]
```diff
@@ -41,7 +41,7 @@ export interface TemplateComponents {
    * NOTE: Variables can be included in the text using curly braces - {{name}}.
    *        Header supports 1 variable
    */
-  header?: TemplateHeader;
+  header?: TemplateHeader | TextHeader;
```
[the remark this answers is not in the document]
Comment (paulkahura) [reply libs/model/convs-mgr/functions/src/lib/base/message-template.interface.ts:44]: here I get an error of tex not existing on header

**Review by ReaganCn — APPROVED** [review]

**Comment by LemmyMwaura** [comment]
CLosing as this was merged internally

**Pull request closed without merging**